slm_lab/agent/memory/onpolicy.py

```diff
@@ -132,9 +132,17 @@ def __init__(self, memory_spec, algorithm, body):
 
     def preprocess_state(self, state, append=True):
         '''Transforms the raw state into format that is fed into the network'''
-        if append:
+        # Deals with the fact that the very first state of an episode is not appended.
+        if self.state_buffer[-1].sum() == 0:
```

Comment:
memory bugs fully fixed in my PR. added an assert checker too. feel free to undo related changes.

Reply:
Thanks! Undone all related changes.